### DISCUSSION TITLE
PHP 8.4 support

### DISCUSSION THREAD

**Discussion by BladeMF** [phpactor/phpactor#2799] [General]
What's the status here? The way I see it there are just two notable features - property setters and getters and the new syntax `new Object()->method()` (without parenthesis).

**Comment by dantleech** [comment]
Basically parser support: https://github.com/microsoft/tolerant-php-parser - although I don't see much activity there so will probably end up cherry-picking to https://github.com/phpactor/tolerant-php-parser

I don't think there will be many if any changes to Phpactor itself.

**Reply by BladeMF** [reply]
I wonder if:
* Setters and getters being a different types of functions will cause changes
* Asymmetric property visibility should affect auto complete when inside/outside a class.
